solutions/swb-reference/integration-tests/tests/multiStep/environment.test.ts

```diff
@@ -70,27 +65,37 @@ describe('multiStep environment test', () => {
       PROJ: expect.anything() // PROJ should be defined
     });
 
-    //Create Environment B
-    console.log('Creating Environment B');
-    const envBBody = {
-      envTypeId: settings.get('envTypeId'),
-      envTypeConfigId: settings.get('envTypeConfigId'),
-      envType: settings.get('envType'),
-      datasetIds: [],
-      name: randomTextGenerator.getFakeText('environment-multistep-test-envB'),
-      description: 'EnvironmentB for multistep environment.test'
-    };
-    const { data: environmentB } = await paSessionEnvironments.create(envBBody, false);
-    expect(environmentB).toMatchObject({
-      id: expect.stringMatching(envUuidRegExp),
-      instanceId: '', // empty string because instanceId value has not been propagated by statusHandler yet
-      provisionedProductId: '', // empty string because provisionedProductId  has not been propagated by statusHandler yet
-      status: 'PENDING',
-      ETC: expect.anything(), //ETC should be defined
-      PROJ: expect.anything() // PROJ should be defined
-    });
+    return environmentA;
+  }
+  /**
+   * This test uses ProjectAdmin1 role to manage EnvironmentA and ITAdmin role to manage EnvironmentB after having been
+   * created by PA1.
+   * ListEnvironments is tested as ITAdmin.
+   * Expect all APIs other than ListEnvironments to work for PA1 role.
+   * Only Start, Stop, Terminate, Get, and ListEnvironments should work for ITAdmin.
+   */
+  test('launch, connect, stop, get, terminate', async () => {
+    const adminSessionEnvironments: Environments = adminSession.resources.projects
+      .project(projectId)
+      .environments();
+    const paSessionEnvironments: Environments = paSession.resources.projects
+      .project(projectId)
+      .environments();
+
+    console.log('Launch Environment A');
+    const environmentA = await launchEnvironment(
+      randomTextGenerator.getFakeText('environment-multistep-test-envA'),
```

Comment:
nit: random text generation can be moved inside the launchEnvironment function.

Reply:
fixed